libraries/SD/src/SD.h

```diff
@@ -84,35 +84,49 @@ class SDClass {
     }
 
     uint8_t type() {
-        return 0;//card.type();
+        sdfs::SDFSImpl* sd = static_cast<sdfs::SDFSImpl*>(SDFS.getImpl().get());
+        return sd->type();
     }
 
     uint8_t fatType() {
-        return 0;//volume.fatType();
+        sdfs::SDFSImpl* sd = static_cast<sdfs::SDFSImpl*>(SDFS.getImpl().get());
+        return sd->fatType();
     }
 
     size_t blocksPerCluster() {
-        return 0;//volume.blocksPerCluster();
+        sdfs::SDFSImpl* sd = static_cast<sdfs::SDFSImpl*>(SDFS.getImpl().get());
+        return sd->blocksPerCluster();
     }
 
     size_t totalClusters() {
-        return 0;//volume.clusterCount();
+        sdfs::SDFSImpl* sd = static_cast<sdfs::SDFSImpl*>(SDFS.getImpl().get());
+        return sd->totalClusters();
     }
 
     size_t blockSize() {
         return 512;
     }
 
     size_t totalBlocks() {
-        return 0;//(totalClusters() / blocksPerCluster());
+        return (totalClusters() / blocksPerCluster());
     }
 
     size_t clusterSize() {
-        return 0;//blocksPerCluster() * blockSize();
+        return blocksPerCluster() * blockSize();
     }
 
     size_t size() {
-        return 0;//(clusterSize() * totalClusters());
+        uint64_t sz = size64();
+	if (sz > (uint64_t)SIZE_MAX) {
+            Serial.printf_P(PSTR("WARNING: SD card size overflow (>= 4GB).  Please update source to use size64().  Returning 0.\n"));
```

Comment:
Should this Serial be hardcoded for output, or a debug mechanism via a macro or whatever?

Reply:
Yes, good point.  Done.